consensus/spos/bls/subroundBlock.go

```diff
@@ -79,6 +87,10 @@ func (sr *subroundBlock) doBlockJob(ctx context.Context) bool {
 		return false
 	}
 
+	if sr.hardforkExclusionHandler.IsRoundExcluded(uint64(currentRound)) {
```

Comment:
can we add a similar condition in the `receivedBlockBodyAndHeader` function just after we decoded the header? We need to prevent the validators from signing a received block if it was decided it is excluded.
```GO
sr.Header = sr.BlockProcessor().DecodeBlockHeader(cnsDta.Header)
```

cc @SebastianMarian do you have a better suggestion?

Reply:
I think here the code will have no effect, only if we want to set an exclusion round in the future, as here the round will be always the current round and not the past/synced round. Also in the validator side, I think we should put this check in processReceivedBlock as this method si called either by receivedBlockBody, receivedBlockHeader or receivedBlockBodyAndHeader, depending on how the proposer has been sent the info. But again, only if the purpose of this code is to avoid something in the future rounds.

Reply:
pushed